src/fp/single_word.rs

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Finite field arithmetic for any field GF(p) for which `p < 2^W`,
+//! where `W` is a specified word size.
+
+/// impl_field_single_word implements field operations for prime modulus
+/// that fits in one word of `W` bits.
```

Comment:
Is this true if `W` is `u128`? The `mul()` looks to be tailored for `u64` (and maybe `u32`).